subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -387,6 +387,12 @@ beforeSettings {
 Using relative file paths to point to Java executables is now deprecated and will become an error in Gradle 9.
 This is done to reduce confusion about what such relative paths should resolve against.
 
+[[test_task_success_without_test]]
+==== Deprecated running the test task successfully without a test present
+Running the `Test` task successfully without a test present is now deprecated and will become an error in Gradle 9.
+This is done to avoid accidental successful test runs due to erroneous configuration.
+The behaviour can be controlled with the task options `--success-without-test` and `--no-success-without-test` or on the `Test` task using link:{javadocPath}/org/gradle/api/tasks/testing/Test.html#setSuccessWithoutTest-java.lang.Boolean-[setSuccessWithoutTest(Boolean)].
```

Comment:
:x: let's link to the `Property<Boolean> getSuccessWithoutTest()` lazy property instead of the command line option setter here.